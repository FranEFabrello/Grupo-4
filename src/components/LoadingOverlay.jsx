import React, { useEffect, useState } from 'react';
import { View, Image, StyleSheet, Dimensions } from 'react-native';
import { BlurView } from 'expo-blur';
import * as Animatable from 'react-native-animatable';
import { useAppTheme } from "~/providers/ThemeProvider";
const { width } = Dimensions.get('window');

import Ionicons from '@expo/vector-icons/Ionicons';

const BASE_SCALE = 2.5;
const SIZE = width * 0.3 * BASE_SCALE;
const LOGO_SIZE = SIZE * 0.5;

const pulseAnimation = {
  0: {
    scale: 1,
    opacity: 0,
  },
  0.2: {
    scale: 1.2,
    opacity: 1,
  },
  0.7: {
    scale: 2.5,
    opacity: 0,
  },
  1: {
    scale: 3,
    opacity: 0,
  },
};

const blueTones = ['#3B82F6', '#60A5FA', '#93C5FD', '#BFDBFE', '#DBEAFE'];

export default function LoadingOverlay() {
  const { colorScheme } = useAppTheme();
  const blurTint = colorScheme === 'dark' ? 'dark' : 'light';

  const [activeHearts, setActiveHearts] = useState([]);

  // Va agregando corazones de a uno
  useEffect(() => {
    blueTones.forEach((_, index) => {
      setTimeout(() => {
        setActiveHearts((prev) => [...prev, index]);
      }, index * 500); // delay entre apariciones
    });
  }, []);

  return (
    <BlurView intensity={50} tint={blurTint} style={styles.blur}>
      <View style={[styles.container, { width: SIZE, height: SIZE }]}>
        {activeHearts.map((i) => (
          <Animatable.View
            key={i}
            animation={pulseAnimation}
            iterationCount="infinite"
            duration={2000}
            useNativeDriver
            style={styles.pulse}
          >
            <Ionicons name="heart-outline" size={LOGO_SIZE * 0.9} color={blueTones[i]} />
          </Animatable.View>
        ))}

        <Image
          source={require('../assets/logo.png')}
          style={{ width: LOGO_SIZE, height: LOGO_SIZE }}
          resizeMode="contain"
        />
      </View>
    </BlurView>
  );
}

const styles = StyleSheet.create({
  blur: {
    position: 'absolute',
    inset: 0,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
  },
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  pulse: {
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
