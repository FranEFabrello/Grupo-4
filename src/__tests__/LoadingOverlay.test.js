jest.mock('expo-blur', () => ({ __esModule: true, BlurView: () => null }));

import React from 'react';
import { render } from '@testing-library/react-native';
import LoadingOverlay from '~/components/LoadingOverlay';

jest.useFakeTimers();

describe('LoadingOverlay', () => {
  it('renders pulse animation and matches snapshot', () => {
    const { toJSON } = render(<LoadingOverlay />);
    jest.advanceTimersByTime(2500);
    expect(toJSON()).toMatchSnapshot();
  });
});