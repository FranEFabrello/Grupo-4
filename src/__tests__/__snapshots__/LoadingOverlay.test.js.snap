// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingOverlay renders pulse animation and matches snapshot 1`] = `null`;
